components/gitpod-cli/cmd/env.go

```diff
@@ -97,14 +98,15 @@ func connectToServer(ctx context.Context) (*connectToServerResult, error) {
 		return nil, xerrors.New("repository info is missing owner")
 	}
 	if wsinfo.Repository.Name == "" {
-		return nil, xerrors.New("repository info is missing name")
+		xerrors.New("repository info is missing name")
 	}
 	repositoryPattern := wsinfo.Repository.Owner + "/" + wsinfo.Repository.Name
 	clientToken, err := supervisor.NewTokenServiceClient(supervisorConn).GetToken(ctx, &supervisor.GetTokenRequest{
 		Host: wsinfo.GitpodApi.Host,
 		Kind: "gitpod",
 		Scope: []string{
-			"function:getEnvVars",
+			"function:getWorkspaceEnvVars",
```

Comment:
TODO: actually compatibility check should happen here, there won't be token otherwise